src/service/xhr-impl.js

```diff
@@ -154,7 +171,7 @@ export class Xhr {
     }
     // For some same origin requests, add AMP-Same-Origin: true header to allow
     // publishers to validate that this request came from their own origin.
-    const currentOrigin = parseUrl(this.win.location.href).origin;
+    const currentOrigin = this.win.location.origin;
```

Comment:
Safe to use? [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Window/location#Browser_compatibility) is unclear.